src/psyclone/psyir/symbols/symboltable.py

```diff
@@ -592,10 +600,11 @@ def remove(self, symbol):
                     symbol.name,
                     [sym.name for sym in self.imported_symbols(symbol)]))
 
-        # If the symbol had any tags, they should be disassociated
+        # If the symbol had a tag, it should be disassociated
         for tag, tagged_symbol in list(self._tags.items()):
             if symbol is tagged_symbol:
                 del self._tags[tag]
+                break
```

Comment:
github isn't allowing me to continue the previous conversation so I have to make a new comment. If the same tag *could* be used by multiple symbols (i.e. we haven't explicitly designed to exclude that option) then I would remove this `break` for safety's sake.

Reply:
Done. I also think is the right thing to do.